lib/git-hooks.js

```diff
@@ -131,14 +146,18 @@ function runHooks(hooks, args, callback) {
         return;
     }
 
-    var hook = spawnHook(hooks.shift(), args);
-    hook.on('close', function (code) {
-        if (code === 0) {
-            runHooks(hooks, args, callback);
-        } else {
-            callback(code);
-        }
-    });
+    try {
```

Comment:
It doesn't suck actually.

Sync version is bad because of two reasons:
  * if a hook works a long time, a user will see the output when the hook ends.
  * You have to specify an enormous size of buffer `maxBuffer` because you don't know how many output the hook prints.